### DISCUSSION TITLE
New hydromt.flw.ldd_from_dem

### DISCUSSION THREAD

**Discussion by AquaPore** [Deltares/hydromt_wflow#299] [Feature request]
A great thanks for providing such a useful tools to derive the flow direction grid from an elevation grid: **hydromt.flw.d8_from_dem**

As I am using the _hydromt_ to prepare the data for **WFLOW** model which requires the flow direction format to be pcraster **LDD** convection. I am wandering if you would have a tool  **hydromt.flw.ldd_from_dem** ?

Thanks for any help you may provide,
Joseph

**Comment by shartgring** [comment]
@AquaPore thank you for asking this question and using Wflow and hydromt. Under the hood, hydromt and hydromt_wflow make use of pyflwdir methods (https://deltares.github.io/pyflwdir/latest/index.html). This package has support for the conversion from d8 to ldd and vice vera, those functions can be found here: https://github.com/Deltares/pyflwdir/blob/main/pyflwdir/core_conversion.py. 

Are you deriving a model for Wflow.jl (https://deltares.github.io/Wflow.jl/dev/) or the old Python-based Wflow (https://wflow.readthedocs.io/en/latest/)? We would recommend sticking to Wflow.jl. Building a Wflow model using hydromt_wflow, both LDD and D8 are supported, see also https://deltares.github.io/hydromt_wflow/latest/_generated/hydromt_wflow.WflowModel.setup_basemaps.html

**Comment by AquaPore** [comment]
A great thanks for pointing me to the right direction. I am using the **Julia Wflow** model .

 It is great to see that there is a package which switches the conversion **d8_to_ldd**(flwdir) &  **ldd_to_d8**(flwdir) may be could add a note in the excellent documentation.

It is great that LDD is supported in hydromt_wflow.

Thanks for your support,
Joseph

**Comment by shartgring** [comment]
Good suggestion Joseph, I will suggest to add it to the documentation: #304

And close this discussion as the question has been answered
